crates/nu-command/src/filesystem/rm.rs

```diff
@@ -263,7 +263,9 @@ fn rm(
                                 continue;
                             }
 
-                            all_targets.entry(f.clone()).or_insert_with(|| target.span);
+                            all_targets
+                                .entry(currentdir_path.join(f))
```

Comment:
This should work, but it is slightly more robust to use `nu_path::expand_path_with(f, &currentdir_path)`. It covers more edge cases, such as joining a path starting with `~` and not ending up with `/current/dir/path/~`.

Reply:
Yep, sounds good.